finetuner/tuner/pytorch/miner.py

```diff
@@ -59,6 +63,99 @@ def mine(
         diffs = matches ^ 1
 
         matches.fill_diagonal_(0)
+
+        triplets = matches.unsqueeze(2) * diffs.unsqueeze(1)
+
+        return torch.where(triplets)
+
+
+class TripletHardMiner(TripletMiner):
```

Comment:
Can you add a docstring for the miner, describing what it does and what the availible strategies are